### PR TITLE
refactor!: add context to the root command  

### PR DESCRIPTION
**Pull request by manu0466**
## Description

This PR adds a new `Context` to the root command generated by the `cmdtypes.RootCmd` function. This new `Context` will provide the parsing configuration, the database instance, the node from which we parse the data, and all the modules registered by the users. With all this data, a client using the Juno library will be able to access the various modules to create custom commands that utilize one or more of these modules.

## Checklist
- [x] Targeted PR against correct branch.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.  
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: types/cmd/context.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+	for _, mod := range allModules {
+		if mod.Name() == module {
+			return mod, nil
+		}
+	}
+
+	return nil, fmt.Errorf("module %s not found", module)
+}
+
+// GetParserContext setups all the things that can be used to later parse the chain state
```
Comment (rustcandy) [inline types/cmd/context.go]:
```suggestion
// GetParseContext setups all the things that can be used to later parse the chain state
```

File: types/cmd/context.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+	// Setup the logging
+	logger, err := ctx.GetLogger()
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the encoding config
+	encodingConfig := ctx.GetEncodingConfig()
+	db, err := ctx.GetDatabase()
```
Comment (rustcandy) [inline types/cmd/context.go:309]:
```suggestion

        // Get database
	db, err := ctx.GetDatabase()
```

**Review by rustcandy — APPROVED** [review]
Overall looks good to me

**Review by rmontagnin — APPROVED** [review]
Fantastic work!

**Pull request merged**